storage/zynq7000-flash/flashdrv.c

```diff
@@ -718,13 +715,39 @@ int flashdrv_devInit(storage_t *strg)
 	strg->dev->blk = malloc(sizeof(storage_blk_t));
 	if (strg->dev->blk == NULL) {
 		resourceDestroy(reg->lock);
-		free(reg->buff);
-		free(strg->dev);
+		free(strg->dev->ctx);
 		free(strg->dev->mtd);
+		free(strg->dev);
 		return -ENOMEM;
 	}
 	strg->dev->blk->ops = &blkOps;
 
+	strg->dev->ctx->cacheCtx = malloc(sizeof(cache_devCtx_t));
+	if (strg->dev->ctx->cacheCtx == NULL) {
+		free(strg->dev->ctx);
+		free(strg->dev->mtd);
+		free(strg->dev->blk);
+		free(strg->dev);
+		return -ENOMEM;
+	}
+
+	strg->dev->ctx->cacheCtx->start = strg->start;
+	strg->dev->ctx->cacheCtx->id = id;
+
+	cacheOps.ctx = strg->dev->ctx->cacheCtx;
+
+	/* cache maps from 0 strg->size */
+	reg->cache = cache_init(strg->size, BLK_CACHE_SECNUM * secSz, secSz, &cacheOps);
```

Comment:
Huh? Why is cache line 4 times sector size instead of sector size? Also, why we want sector size number of lines? I don't see connection between sector size in bytes and number of lines tbh.

Reply:
Oh no, it was using the old function prototype. Fixed it, thank you.